sensor/stmemsc/README

```diff
@@ -25,18 +25,26 @@ Description:
 
    The driver is platform-independent, you only need to define the two
    functions for read and write transactions from the sensor hardware bus
-   (ie. SPI or I2C).
+   (ie. SPI or I2C). In addition you may define a mdelay (milliseconds) 
+   routine.
 
    Define in your 'xyz' driver code the read and write functions that use the
-   I2C or SPI platform driver like the following:
+   I2C or SPI platform driver (plus the optional mdelay function) like the following:
 
    /** Please note that is MANDATORY: return 0 -> no Error.**/
    int platform_wr(void *handle, u8_t reg, const u8_t *bufp, u16_t len);
    int platform_rd(void *handle, u8_t reg, u8_t *bufp, u16_t len);
 
+   /** Component optional fields **/
+   void platform_mdelay(u32_t millisec);
```

Comment:
I guess every user interested in that functionality would need the same zephyr porting function of this.
Can we think of a way to provide it in Zephyr directly, so people don't have to implement it on their side?

Reply:
> I guess every user interested in that functionality would need the same zephyr porting function of this. Can we think of a way to provide it in Zephyr directly, so people don't have to implement it on their side?

Good idea. 
What I can do is to add stmemsc_delay() in file https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/stmemsc/stmemsc.h:

```
void platform_mdelay(uint32_t ms)
{
  k_msleep(ms);
}
```

 which can then be used in the stmemc based drivers (like here https://github.com/zephyrproject-rtos/zephyr/blob/1f57517e2c5fc923686883781838c3f6fdbec978/drivers/sensor/lsm6dso/lsm6dso.c#L955)

What do you think?

Reply:
That would be great, but only thing is that can't do it that way. Porting on zephyr symbols should be done in zephyr main repo.
See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/common/stm32cube_hal.c

Reply:
> That would be great, but only thing is that can't do it that way. Porting on zephyr symbols should be done in zephyr main repo. See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/common/stm32cube_hal.c

Sure, I meant that I'd do it in zephyr main repo. Seems very similar to HAL_Delay() implementation infact.